doc/src/manual/style-guide.md

```diff
@@ -119,6 +119,10 @@ with both copying and modifying forms (e.g., [`sort`](@ref) and [`sort!`](@ref))
 which are just modifying (e.g., [`push!`](@ref), [`pop!`](@ref), [`splice!`](@ref)).  It
 is typical for such functions to also return the modified array for convenience.
 
+Functions related to IO or making use of random number generators (RNG) are notable exceptions:
+Since these functions almost invariably must mutate the IO or RNG, respectively, functions ending with `!` are used to signify that an argument _other_ than the IO or RNG is mutated.
```

Comment:
```suggestion
Since these functions almost invariably must mutate the IO or RNG, respectively, functions ending with `!` are used to signify a mutation _other_ than mutating the IO or incrementing the RNG state.
```
Strange that `close(io)` is not marked similarly to `Random.seed!`—they both perform non-standard mutations to their argument.